[c10d] add more fields for periodic logging

shuqiangzhang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #123860

Summary:
Added the names of the last enquened, started and completed colletives,
in addition to their seq ID
Test Plan:
CI

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang

shuqiangzhang committed:
[c10d] add more fields for periodic logging
Summary:
Added the names of the last enquened, started and completed colletives,
in addition to their seq ID
Test Plan:
CI

[ghstack-poisoned]

XilunWu commented on torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:
```diff
@@ -1647,9 +1653,18 @@ void ProcessGroupNCCL::watchdogHandler() {
+        lastStartedSeq_ = work.seq_;
+        lastStartedWorkName_ = opTypeToString(work.opType_);
+      }
+
```
n00b question: will `work.seq_` always be checked in the numeric order? i.e. the collective work with `seq_=5` won't be checked until `seq_=4` is checked.

XilunWu reviewed (approved):
lgtm! left 1 basic question.

shuqiangzhang committed:
Update on "[c10d] add more fields for periodic logging"


Summary:
Added the names of the last enquened, started and completed colletives,
in addition to their seq ID
Test Plan:
CI

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

shuqiangzhang commented:
@pytorchbot merge
